server/driver/wivrn_comp_target.cpp

```diff
@@ -596,6 +610,25 @@ void wivrn_comp_target::reset_encoders()
 	cnx->send_control(desc);
 }
 
+void wivrn_comp_target::render_dynamic_foveation(std::array<to_headset::foveation_parameter, 2> foveation)
+{
+	assert(foveation_renderer);
+
+	vk::SubmitInfo submit_info{
+	        .commandBufferCount = 1,
+	        .pCommandBuffers = &*foveation_renderer->cmd_buf,
+	};
+
+	foveation_renderer->render_distortion_images(foveation, c->nr.distortion.images, c->nr.distortion.image_views);
+
+	wivrn_bundle->device.resetFences(*foveation_renderer->fence);
+	{
+		scoped_lock lock(c->base.vk.queue_mutex);
+		wivrn_bundle->queue.submit(submit_info, *foveation_renderer->fence);
+	}
+	auto res = wivrn_bundle->device.waitForFences(*foveation_renderer->fence, true, UINT64_MAX);
```

Comment:
Isn't there a way to include this in the command buffer used by monado? We introduce a GPU-CPU synchronization point

Reply:
i ended up keeping the separate cmd buf, but dropped the fence